Added a global event handler for beforeinput

phistuck opened the pull request:
Also imported the input event from the UI events
specifications, since that one defines it.

Fixed https://github.com/whatwg/html/issues/4526


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***

### :boom: Error: Wattsi server error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Jan 15, 2021, 7:59 AM UTC)_.

<details>
<summary>More</summary>


PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [Wattsi Server](https://github.com/domenic/wattsi-server) - Wattsi Server is the web service used to build the WHATWG HTML spec.

:link: [Related URL](https://build.whatwg.org/wattsi)

```
<html>
<head><title>504 Gateway Time-out</title></head>
<body bgcolor="white">
<center><h1>504 Gateway Time-out</h1></center>
<hr><center>nginx/1.10.3</center>
</body>
</html>

```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20whatwg/html%234530.)._
</details>

sideshowbarker commented:
heads-up @whatwg/documentation

annevk commented on source:
```diff
@@ -121168,6 +121172,12 @@ INSERT INTERFACES HERE
+    <tr> <!-- beforeinput -->
+     <td> <code data-x="event-beforeinput">beforeinput</code>
+     <td> <code>Event</code>
+     <td> Form controls
```
It fires at more than form controls, no?

phistuck replied:
Same for `input`, though. Is `contenteditable` considered as a form control (I guess not)?
How would you word it if `contenteditable` needs to be included?
Besides, this columns is named "Interesting targets" and it is non-normative, I am not sure it is exhaustive anyway.

annevk replied:
I'd make it "Elements", similarly to copy and cut and such.

phistuck replied:
@annevk -
Done for `input` as well.

annevk replied:
So the description also needs changes. Maybe "Fired when the user edits text, before the text is changed" and similar for input?

phistuck replied:
I think they fire before/when you paste or delete an image, which is not necessarily just text, so I opted for `content/value` (for `contenteditable` elements and form controls).
I do not feel strongly about it, if you insist.

chrisdavidmills commented:
Documentation potential need recorded at https://trello.com/c/aBs1o6Pg/86-dom-general-enhancements-fx-67

phistuck commented:
@annevk - and verifying my participation.
Looks like WebKit added it already, without adding it to the standard -
https://trac.webkit.org/changeset/206944/webkit#file12
They added it to `Element` and `Document`, rather than to `GlobalEventHandlers` -
https://github.com/WebKit/webkit/blob/master/Source/WebCore/dom/DocumentAndElementEventHandlers.idl
They also seem to have added support for it as a content attribute (should new events be added as content attributes? I assumed not) -
https://github.com/WebKit/webkit/blob/master/Source/WebCore/html/HTMLAttributeNames.in

Chromium issue - https://bugs.chromium.org/p/chromium/issues/detail?id=947408

Intention by Mozilla to add it when they implement `beforeinput` -
https://github.com/w3c/uievents/issues/218#issuecomment-477899033

phistuck commented:
@annevk - regarding tests, yes, I assumed IDLs are automatically tested.

annevk commented:
Ah yeah, it should be a content attribute as well.

phistuck commented:
@annevk - I will add it in the UTC+2 evening. Anything else left other than the content attribute and verifying my participation?

annevk commented:
No, I think we're good then, thanks! I'll verify you now.

phistuck commented:
> Ah yeah, it should be a content attribute as well.

Unrelated, but why does `DOMContentLoaded` not have an IDL/content attribute, too?

phistuck commented:
@annevk -

> Ah yeah, it should be a content attribute as well.

Unrelated, but why does `DOMContentLoaded` not have an IDL/content attribute, too?

I added the content attribute.

One more thing, is adding my name to the acknowledgement section mandatory? I assumed it was, so I added, but I generally rather not if it is possible.

domenic commented:
I'm not sure this should be verified, as I don't believe signing a contract using a pseudonym is viable legally...

phistuck commented:
@domenic - Google was fine with it, jQuery was fine with it and maybe a few more I signed (RavenDB was not fine but only because they required an exact address and phone number which I am not willing to provide). But if you have an issue with that, I hereby release all rights for my change and you or anyone else is welcome to take it as their own, no credit needed.

annevk commented:
@domenic I've verified other pseudonym-like names. Also, without ID verification of some kind many others could be pseudonyms too?

annevk commented:
@phistuck adding your name to the Acknowledgments section is optional, feel free to remove it. `DOMContentLoaded` could maybe have an event handler attribute too. I think historically we might have tried to steer folks away from event handler attributes, but there's no real good reason to do so. One problem there is that the event name is not lowercase, but the event handler needs to be. That'd be a first and might be problematic for some implementations. Best discussed separately.

phistuck commented:
@annevk - done.

Thank you for the review, everyone.

phistuck commented:
@annevk and @domenic - is there anything else that is needed for this to be merged?

Like I mentioned, if there is a legal issue here, feel free to take the change as your own, credit-attribution-acknowledgement free, I am waiving all rights.

phistuck commented:
@annevk and @domenic - from the agreement status page -
> After an agreement is submitted, it is verified by the WHATWG. This is a manual process, but is usually completed quickly if it's preventing a pull request from being accepted.

This is preventing a pull request from being accepted for three months. It would be great if you expedited the process.

domenic commented:
We are blocked on confirmation from the SG as to whether pseudonyms are acceptable. This is out of our hands. We can close the PR if that's preferable to you.

mfreed7 commented:
Any update on this and https://github.com/whatwg/html/issues/4526? Seems a shame to not be able to land the PRs over names vs. pseudonyms. Can we land this under a different name, or does a fresh PR need to be made?

domenic commented:
A fresh PR is probably the best way forward.

phistuck commented:
@mfreed7 - sure, go for it, let me know when you have and I will close this one.

mfreed7 commented:
> @mfreed7 - sure, go for it, let me know when you have and I will close this one.

https://github.com/whatwg/html/pull/6743
